### ISSUE TITLE
Different Dice accuracy  using DataParallel 

### ISSUE THREAD

**Issue by zs963048949** [pytorch/pytorch#49794]
## ❓ Questions and Help
Hello!!! I got a problem in DataParallel multiple gpu training. I got different value of segmentation accuracy,when I used different numbers of GPUs in the way of DataParallel.The difference values between any two dice accuracy differ from 2% to 4%.What resulted it?

**Comment by zs963048949** [comment]
I have a supplement.I have set all random seed.Besides,I have eliminated all other impact about random.The program have only one difference ,and it is the number of GPU.

**Comment by SsnL** [comment]
do you use batch norm?

**Comment by zs963048949** [comment]
> do you use batch norm?

yes，it was not  matter.The accuracy is euqal when I run three the same train code in one or two GPU.

**Comment by zs963048949** [comment]
> do you use batch norm?

Yes,and I realized just that it seems to have a difference for the BN. However,why did it cause this problem?

**Comment by SsnL** [comment]
usual bn normalizes per device. you might want to use synchronous bn instead if you want global norm.

**Comment by zs963048949** [comment]
> usual bn normalizes per device. you might want to use synchronous bn instead if you want global norm.
Do you know whether DP and DDP have the same problem? Thanks in advance!

**Comment by SsnL** [comment]
It doesn't matter if it is DP or DDP. You need to use Sync BN if you need global norm.

**Comment by zs963048949** [comment]
Thanks，I got it. Do you know some good methods about Sync BN? Thanks in advance.



---Original---
